### ISSUE TITLE
remote write 2.0 - decide whether addition of metadata should count towards max samples in write request

### ISSUE THREAD

**Issue by cstyan** [prometheus/prometheus#14407]
### Proposal

in queue manager append functions we pass the batch to the shard to be sent when it has reached the configured `max_sample_per_send`, we decided that exemplars would be counted but that we'd increase the max allowed samples by a multiple

we should decide whether or not to do the same with metadata, or not count metadata as a part of the max samples in the request

**Comment by bboreham** [comment]
Hello from the bug-scrub!  Looks like nobody had an opinion to share here.

Given that the code today must embody a choice, could we just document that as the answer and close this issue?

**Comment by cstyan** [comment]
@bboreham yes, where do you think is the correct place to document this? Just in the documentation for the `max_samples_per_send` queue config option?

**Comment by bwplotka** [comment]
IMO it should NOT count, main reasons is implication of https://github.com/prometheus/prometheus/issues/15911 and switching to type-and-unit labels. Let's document it and close this issue. Option documentation feels ok.

**Comment by bwplotka** [comment]
Metadata is not a sample in the current RW2 spec, so let's not add it.
